### ISSUE TITLE
`comma-dangle` warning for destructuring assignment when skipping the last entry

### ISSUE THREAD

**Issue by philer** [eslint/eslint#7654]
(I'll take the liberty of restructuring your template as it makes no sense to describe the actual problem below my entire configuration…)

Destructuring array assignment offers the possibility to [skip/ignore values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Ignoring_some_returned_values). In my particular use case I want to **skip the *last* entry** of an array with a fixed length of 2.

Easiest way to reproduce:
```js
const [a,] = [1,2]; // warning  Unexpected trailing comma  comma-dangle
```
My personal use case (part of a minimalistic Set implementation):
```js
contains() {
  const [found,] = this.search(x); // returns an array like [true, 5] or [false, -1]
  return found;
}
```
**Note:** I am aware that I could use `return this.search(x)[0];` instead. However I still believe that this should be possible to use as expected and I assume there will be other use cases that actually require it.

**Note 2:** Obviously this common pattern from other languages doesn't work either: 
```const [a,_] = [1,2]; // error    '_' is defined but never used  no-unused-vars```

**What did you expect to happen?**
ESLint doesn't complain as this dangling comma actually has meaning.

**What actually happened? Please include the actual, raw output from ESLint.**
```  54:17  warning  Unexpected trailing comma  comma-dangle```

------------------

* **ESLint Version:** v3.6.1
* **Node Version:** v6.9.1
* **npm Version:** 3.10.8

* **Using `babel-parser`**.

`.eslintrc.json`:
```json
{
  "root": true,
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "block-scoped-var": "warn",
    "comma-dangle": [
      "warn",
      "always-multiline"
    ],
    "no-console": "off",
    "no-var": "warn",
    "semi": [
      "error",
      "always"
    ],
    "strict": "error",
    "no-cond-assign": "off"
  },
  "env": {
    "browser": true
  },
  "globals": {
    "require": true,
  }
}
```

**Comment by not-an-aardvark** [comment]
Thanks for the issue. However, I don't think I understand what the problem is. Regardless of how long the array actually is, the dangling comma isn't doing anything, so you can just omit it:

```js
contains() {
  const [found] = this.search(x); // returns an array like [true, 5] or [false, -1]
  return found;
}
```

**Comment by philer** [comment]
I confess that skipping the comma did not even occur to me. I assumed that for destructuring assignments the semantics would be different and the comma required to indicate a trailing empty slot. 
Thanks for clearing it up!
